### PR TITLE
add new scenario for terraform consul provider guide

### PR DESCRIPTION
**Pull request by im2nguyen**
Scenario for the Terraform Consul Provider guide 

Link: https://deploy-preview-1122--hashicorp-learn.netlify.com/consul/developer-discovery/terraform-consul-provider

### CODE REVIEW

File: terraform-consul-provider/overview.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+This is the workspace for the [**Register External Services with Terraform**](http://learn.hashicorp.com/consul/developer-discovery/terraform-consul-provider) guide.
+
+These services can be open by clicking on their respective tabs in the terminal.
+
+On initialization, this workspace should have ran `docker-compose up -d` which spins
+up the above services in the background. To view these instances, run `docker ps`{{execute}}
+
+It takes about a minute after you see the `Ready` message for the Consul datacenter
+to nominate a cluster leader and for you to continue to the next step.
```
Comment (judithpatudith) [inline terraform-consul-provider/overview.md]:
```suggestion
On initialization, this workspace should have ran `docker-compose up -d` creates all the services listed in the Learn guide. To view these instances, run `docker ps`{{execute}}

It takes about a minute after you see the `Ready` message for the Consul datacenter
to nominate a cluster leader and for you to continue to the next step.

Continue following the instructions in the learn guide for bootstrapping Consul's ACL system. 
```

File: terraform-consul-provider/overview.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+## Bootstrap Consul datacenter
+
+`docker exec -it consul-playground_consul-server-1_1 consul acl bootstrap`{{execute}}
+
+For more information about this command, please reference the guide.
+
```
Comment (judithpatudith) [inline terraform-consul-provider/overview.md]: I think we can remove this bit since the guide covers it

Comment (im2nguyen) [reply terraform-consul-provider/overview.md]: I think it's nice to click on the command to bootstrap Consul vs typing/copying and pasting it into the terminal

File: terraform-consul-provider/overview.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+## Localhost Substitution
+
+NOTE: If you are using Katacoda, Katacoda uses a unique method to route its services.
+
+As a result, you should substitute any references to localhost to the following:
```
Comment (judithpatudith) [inline terraform-consul-provider/overview.md]:
```suggestion
Katacoda uses a unique method to route its services. As you follow the guide substitute any references to localhost with the following addresses, in order to make the configuration work with KataCoda.
```

File: terraform-consul-provider/overview.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+| Learn Address   | Katacoda Address |
+| --------------  | ---------------- |
+| localhost:8500  | `[[HOST_SUBDOMAIN]]-8500-[[KATACODA_HOST]].environments.katacoda.com` |
+| localhost:9001  | `[[HOST_SUBDOMAIN]]-9001-[[KATACODA_HOST]].environments.katacoda.com` |
+| localhost:8080  | `[[HOST_SUBDOMAIN]]-8080-[[KATACODA_HOST]].environments.katacoda.com` |
+
+Remember to substitute `ACL_TOKEN_HERE` with your master ACL token. Your `main.tf` file should be similar to:
```
Comment (judithpatudith) [inline terraform-consul-provider/overview.md]: Alternatively you can copy the following `main.tf` file, which represents the finished state of the guide and has the correct addressing. Remember to substitute `ACL_TOKEN_HERE` with your master ACL token. 

If you use this file you will only need to run `terraform apply`{{execute}} once to see all the outputs described in the guide.

File: terraform-consul-provider/overview.md (hunk 2 above)
Comment (judithpatudith) [reply terraform-consul-provider/overview.md]: 👌 in that case lets just tell people where they should hop back to the guide. (move the current line 10 and have it describe the step they should jump back into)

Comment (im2nguyen) [reply terraform-consul-provider/overview.md]: Not sure what you meant by current line 10... but I think I resolved it?

```
It takes about a minute after you see the `Ready` message for the Consul datacenter to nominate a cluster leader and for you to continue to the next [step](http://learn.hashicorp.com/consul/developer-discovery/terraform-consul-provider#bootstrap-consul-acls), bootstrapping your Consul ACLs.

## Bootstrap Consul ACLs

This corresponds to the [Bootstrap Consul ACLs](http://learn.hashicorp.com/consul/developer-discovery/terraform-consul-provider#bootstrap-consul-acls) section on the Learn guide. Once you run the following command, use the guide to complete the rest of this scenario.

`docker exec -it consul-playground_consul-server-1_1 consul acl bootstrap`{{execute}}
```

**Pull request merged**